parser/src/ast/integrity_constraints.rs

```diff
@@ -7,8 +7,11 @@ use super::{EvaluatorFunctionCall, Expression, VariableBinding};
 
 #[derive(Debug, Eq, PartialEq)]
 pub enum IntegrityStmt {
-    Constraint(ConstraintType, Option<Expression>),
-    ConstraintComprehension(ConstraintType, Option<Expression>, ComprehensionContext),
+    Constraint(
+        ConstraintType,
+        Option<Expression>,
+        Option<ComprehensionContext>,
+    ),
```

Comment:
I think maybe we should make a new struct to hold this. Maybe something like the following:

```
#[derive(Debug, Eq, PartialEq)]
pub enum IntegrityStmt {
    Constraint(IntegrityConstraint),
    VariableBinding(VariableBinding),
}

#[derive(Debug, Eq, PartialEq)]
pub struct IntegrityConstraint {
    constraint_expr: ConstraintExpr,
    selectors: Option<Expression>,
    comprehension_context: Option<ComprehensionContext>,
}

#[derive(Debug, Eq, PartialEq)]
enum ConstraintExpr {
    Inline(InlineConstraintExpr),
    Evaluator(EvaluatorFunctionCall),
}

/// Stores the expression corresponding to the integrity constraint.
#[derive(Debug, Eq, PartialEq, Clone)]
struct InlineConstraintExpr {
    lhs: Expression,
    rhs: Expression,
}
```

Reply:
I was considering doing this as this does look cleaner but decided against it since `IntegrityConstraint` seems like it should be a single constraint and constraint comprehension seems like syntactic sugar to write multiple constraints in a single statement. Although I don't have a strong opinion on this so happy to change it if you prefer what you suggested instead.

Reply:
I think taking an approach like this will make the IR cleaner, so I would prefer to change it

>constraint comprehension seems like syntactic sugar to write multiple constraints in a single statement

Maybe this can be addressed just by using different names for the structs above. For example, instead of `IntegrityConstraint` it could be `ConstraintStmt` or something like that

Reply:
I went with the structure you suggested.